lib/class-vip-request-block.php

```diff
@@ -137,11 +144,29 @@ public static function block_and_log( string $value, string $criteria ) {
 			header( 'Expires: Wed, 11 Jan 1984 05:00:00 GMT' );
 			header( 'Cache-Control: no-cache, must-revalidate, max-age=0' );
 
-			// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
-			error_log( sprintf( 'VIP Request Block: request was blocked based on "%s" with value of "%s"', $criteria, $value ) );
+			if ( static::$should_log ) {
+				// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+				static::log( $criteria, $value );
+			}
+
 			exit;
 		}
 
 		return true;
 	}
+
+	public static function log( string $criteria, string $value ): void {
+		// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+		error_log( sprintf( 'VIP Request Block: request was blocked based on "%s" with value of "%s"', $criteria, $value ) );
+	}
+
+	/**
+	 * Control logging of blocked requests.
+	 *
+	 * @param bool $should_log whether to log blocked requests.
+	 * @return void
+	 */
+	public static function should_log( bool $should_log = true ) {
```

Comment:
Minor nitpick, but I'm not a huge fan of the function name...I feel like we should rename it, since at first glance, `should_log()` reads to me like a function that returns on what the current logging status is, rather than that it controls it. Maybe a bit more tedious, but how would you feel about having two explicit functions, `enable_logging_blocked_requests()` and `disable_logging_blocked_requests()`?

Reply:
I like the suggestion for `enable_logging()` and `disable_logging()` because it's so clear and human-readable. It's also similar to how PHPCS scanning is enabled and disabled for sections of code (`phpcs:enable` and `phpcs:disable`).